pkg/mesos/taskio.go

```diff
@@ -287,9 +287,17 @@ func (t *TaskIO) attachContainerInput() error {
 			}()
 
 			for {
-				// TODO(bamarni): investigate on why "test_task:test_attach" fails if the buffer size below
-				// is greater than 1, this might be an issue with the term package or the way we use it.
-				buf := make([]byte, 1)
+				// Use a buffer with a reasonable size in order to perform well when STDIN is piped and has
+				// a significant size. See https://jira.mesosphere.com/browse/DCOS_OSS-5357
+				bufLen := 512
+
+				if t.opts.TTY {
+					// TODO(bamarni): investigate on why "test_task:test_attach" fails if the buffer size below
```

Comment:
is there a JIRA for this TODO? if so, mind linking it to this PR?

Reply:
There is none, on the other hand I'm not sure we should have one. As there is no known bug or performance issue because of this, I don't see how it would make it to one of our sprint.

As far as I could tell when testing manually it is reading byte by byte when the terminal is in raw mode anyways. The issue was only seen on this integration test with a pty, which made the CLI read more than 1 byte at a time: https://github.com/dcos/dcos-core-cli/blob/6c5d73f51087aee9527b03464f6b91719ab87100/python/lib/dcoscli/tests/integrations/test_task.py#L437-L438